### PR TITLE
adds metric for session_details view

### PR DESCRIPTION
**Pull request by Jtang-1**
## Product Description

Adds metric for Datadog [widget](https://app.datadoghq.com/dashboard/h78-9qs-7cx/ush-performance-monitoring?fullscreen_end_ts=1704494564189&fullscreen_paused=false&fullscreen_refresh_mode=sliding&fullscreen_section=overview&fullscreen_start_ts=1704480164189&fullscreen_widget=3530931766462588&refresh_mode=sliding&tpl_var_environment%5B0%5D=staging&view=spans&from_ts=1704478561648&to_ts=1704492961648&live=true) 

## Technical Summary

[USH-4016](https://dimagi-dev.atlassian.net/browse/USH-4016?focusedCommentId=304392&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-304392)

Investigation of slow load times for validating case search inputs found `/session_details` request as a potential bottleneck. There were a couple Datadog traces that pointed to this. One such example [trace](https://app.datadoghq.com/apm/traces?query=%40_top_level%3A1%20resource_name%3A%22POST%20%2Fnavigate_menu%22%20%40domain%3Aco-carecoordination-test&cols=service%2Cresource_name%2C%40duration%2C%40http.status_code%2C%40_duration.by_service%2C%40domain%2C%40user&graphType=flamegraph&historicalData=false&query_translation_version=v0&saved-view-id=1271135&shouldShowLegend=true&sort=time&spanID=7110734751025576253&spanType=service-entry&spanViewType=metadata&timeHint=1704326482967&trace=60196447845999155917110734751025576253&traceID=6019644784599915591&traceQuery=&view=spans&viz=stream&start=1704325603173&end=1704326503173&paused=false)
This metric will be used to track request timings for `session_details` request on DD and see if any spikes there correspond to the behavior we are seeing with users.
 
## Feature Flag

N/A
### Safety story

tested on staging

### Automated test coverage

no automated test
### QA Plan

no QA

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

[USH-4016]: https://dimagi-dev.atlassian.net/browse/USH-4016?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by AddisonDunn — APPROVED** [review]
I'm not super familiar with DD stuff, so maybe you want to wait for input from someone more experienced, but this looks harmless & seems useful to me.

File: corehq/apps/hqwebapp/session_details_endpoint/views.py [hunk 1]
```diff
@@ -69,6 +72,14 @@ def post(self, request, *args, **kwargs):
...
 
         enabled_toggles = toggles_enabled_for_user(user.username) | toggles_enabled_for_domain(domain)
+        enabled_previews = previews_enabled_for_domain(domain)
+        end_time = datetime.now()
+        metrics_histogram("commcare.session_details.processing_time",
+                      int((end_time - start_time).total_seconds() * 1000),
+                      bucket_tag='duration_bucket',
+                      buckets=(100, 250, 500, 1000, 2000, 3000, 4000),
+                      bucket_unit='ms',
+                      tags={'domain': domain})
```
Comment (esoergel) [inline corehq/apps/hqwebapp/session_details_endpoint/views.py:82]: Would you mind throwing in `limit_domains` here?
https://github.com/dimagi/commcare-hq/blob/f51b264bfbe167b0440e67da464639bfa7e54131/corehq/util/metrics/__init__.py#L317-L321

File: corehq/apps/hqwebapp/session_details_endpoint/views.py [hunk 2]
```diff
@@ -69,6 +72,14 @@ def post(self, request, *args, **kwargs):
             domains.update(EnterprisePermissions.get_domains(member_domain))
 
         enabled_toggles = toggles_enabled_for_user(user.username) | toggles_enabled_for_domain(domain)
+        enabled_previews = previews_enabled_for_domain(domain)
+        end_time = datetime.now()
+        metrics_histogram("commcare.session_details.processing_time",
+                      int((end_time - start_time).total_seconds() * 1000),
+                      bucket_tag='duration_bucket',
+                      buckets=(100, 250, 500, 1000, 2000, 3000, 4000),
```
Comment (esoergel) [inline corehq/apps/hqwebapp/session_details_endpoint/views.py:80]: like the `limit_domains` comment - we get charged based on the number of unique tag combinations used, and datadog histograms actually turn buckets into tags, so using 7 divisions can actually be significant.  Is there a smaller number of splits that would still be useful?

Comment (Jtang-1) [reply corehq/apps/hqwebapp/session_details_endpoint/views.py]: Ah, that's good to know. [fa1885c](https://github.com/dimagi/commcare-hq/pull/33955/commits/fa1885c1e5b3bd5ee052a27213b9e451700610a6)

Based on what i've seen on the past 3 days on staging, 3 buckets would be enough. 
0 - 250ms: fast
250ms - 1 sec: med slow
1sec - 3 sec: slow
Screen Shot 2024-01-08 at 6 42 59 PM

File: corehq/apps/hqwebapp/session_details_endpoint/views.py (hunk 1 above)
Comment (Jtang-1) [reply corehq/apps/hqwebapp/session_details_endpoint/views.py]: [fa1885c](https://github.com/dimagi/commcare-hq/pull/33955/commits/fa1885c1e5b3bd5ee052a27213b9e451700610a6)

**Review by esoergel — APPROVED** [review]

**Pull request merged**